uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/App.java

```diff
@@ -84,20 +91,43 @@ public App(@NonNull final String packageName,
         this.appInstaller = appInstaller;
         this.packageName = packageName;
         this.appName = appName;
+        // update installer is PlayStore by default
+        this.updateAppInstaller = new LocalApkInstaller();
+    }
+
+    public App(@NonNull final String packageName,
+               @NonNull final String appName,
+               @NonNull final IAppInstaller appInstaller,
+               @NonNull final IAppInstaller updateAppInstaller) {
+        this.appInstaller = appInstaller;
+        this.packageName = packageName;
+        this.appName = appName;
+        this.updateAppInstaller = updateAppInstaller;
     }
 
     @Override
     public void install() {
-        //TODO: make it build time configurable to specify the installer that should be used.
-        // Ideally we can specify different installers on app basis
-        if (appInstaller instanceof LocalApkInstaller && !TextUtils.isEmpty(localApkFileName)) {
-            Logger.i(TAG, "Installing the " + this.appName + " from local apk..");
-            appInstaller.installApp(localApkFileName);
-        } else {
-            Logger.i(TAG, "Installing the " + this.appName + " from Play Store..");
-            appInstaller.installApp(packageName);
+        if (localOldApkFileName != null) {
+            // To ensure to install the old apk first in update scenarios
+            if (appInstaller instanceof LocalApkInstaller && !TextUtils.isEmpty(localOldApkFileName)) {
+                Logger.i(TAG, "Installing the " + this.appName + " from local apk..");
+                appInstaller.installApp(localOldApkFileName);
+            } else {
+                Logger.i(TAG, "Installing the " + this.appName + " from Play Store..");
+                appInstaller.installApp(packageName);
+            }
+        }
+        else {
+            //TODO: make it build time configurable to specify the installer that should be used.
+            // Ideally we can specify different installers on app basis
+            if (appInstaller instanceof LocalApkInstaller && !TextUtils.isEmpty(localApkFileName)) {
+                Logger.i(TAG, "Installing the " + this.appName + " from local apk..");
+                appInstaller.installApp(localApkFileName);
+            } else {
+                Logger.i(TAG, "Installing the " + this.appName + " from Play Store..");
+                appInstaller.installApp(packageName);
+            }
```

Comment:
I think you can consider reducing the number of `if/else` blocks here. As a hint, null check and `TextUtils.isEmpty` check are similar with the latter being a more comprehensive check